low-level/framework/utils/ipmi_client.py

```diff
@@ -252,6 +252,50 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
 
         return out, error, retcode
 
+    def load_server_fru_list(self):
+        """Get Server FRU list and merge it with server declare fru list
+        maintained in config, with which FRU list can be extended
+        for a solution.
+        
+        Ex: Supermicro servers not listing disk as FRU,
+        though its most common FRU in servers, and 
+        practically it can be replaced easily. 
+        So if for a solution, FRU list needs to be extended
+        beyond what server publishes, 'server_declare_fru' config in sspl.conf
+        can be used. 
+        Some of the usual FRU examples are:- disk, psu, fan."""
+
+        cmd = 'fru list'
+        self.fru_list = []
+        output, err, rc = self._run_ipmitool_subcommand(cmd,
+        grep_args="FRU Device Description : ")
+        if rc != 0:
+            logger.error("Failed in fetching FRU info from server."
+                         f"Error:{err}")
+        if output:
+            for line in output.split('\n'):
+                self.fru_list.append(line.split(': ')[1])
+            keywords = ['Pwr Supply', 'power', 'PS', 'PSU']
+            for match in keywords:
+                for fru in self.fru_list:
```

Comment:
@cdeshmukh  two for loops will be required, this is not intersection of two list , it is more like pattern matching from list1 into list2.
For eg:-
list1 = ['Pwr Supply', 'power', 'PS', 'PSU']                             // patterns to match
list2 = ['Pwr Supply 1 FRU (ID 2)', 'fan', 'PS1', 'PS1 (ID 1)']   // this the server list we get from ipmitool

so, one for loop will be required to traverse the patterns and second for loop to match pattern str in every element present in list2.

if ''Pwr Supply' in 'Pwr Supply 1 FRU (ID 2)':
replace 'Pwr Supply 1 FRU (ID 2)' with 'psu'

something like this

Reply:
@gargpalak048 you can use regexp. Based on your example above, you can use something like this:
```
import re

list1 = ['Pwr Supply', 'power', 'PS', 'PSU']
list2 = ['Pwr Supply 1 FRU (ID 2)', 'fan', 'PS1', 'PS1 (ID 1)']

fru_regex = re.compile("|".join(list1))

for fru in list2:
    if fru_regex.search(fru):
        list2[list2.index(fru)] = 'psu'
```
```
>>> print(list2)
['psu', 'fan', 'psu', 'psu']
```

Reply:
yes regex we can.
Incorporated the changes @jjxsg

Reply:
But I think regex.search also internally traverse through list.
It is like another for loop only

Reply:
For loop is native python loop but `re` behind the scenes is implemented in C so it's a bit faster. Although here, it will not make any significant difference in terms of performance since size of list is small.